File: frontend/src/app/shared/dto/requests/workouts/log-workout-request.ts
import { ExerciseType } from '../../../models/exercise-type';

export interface LogWorkoutRequest {
  exerciseType: ExerciseType;
  durationMinutes: number;
  caloriesBurned: number;
  difficulty: number;
  fatigue: number;
  notes?: string;
  performedAt: string;
}

File: backend/Api/src/WorkoutApp.Application/DTOs/Workout/LogWorkout/LogWorkoutRequest.cs
using WorkoutApp.Domain.Enums;

namespace WorkoutApp.Application.DTOs.Workout.LogWorkout;

public sealed record LogWorkoutRequest(
    ExerciseType ExerciseType,
    int DurationMinutes,
    int CaloriesBurned,
    int Difficulty,
    int Fatigue,
    string? Notes,
    DateTime PerformedAt);